src/main/java/org/fogbowcloud/manager/core/ManagerController.java

```diff
@@ -1,85 +1,44 @@
-		this.attendOpenOrdersExecutor = new ManagerScheduledExecutorService(Executors.newScheduledThreadPool(1));
+	private String localMemberId;
 
-		this.scheduleExecutorsServices(properties);
-	}
+	private static final Logger LOGGER = Logger.getLogger(ManagerController.class);
 
-	private void scheduleExecutorsServices(Properties properties) {
-		if (!this.attendOpenOrdersExecutor.isScheduled()) {
-			String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SCHEDULER_PERIOD_KEY,
-					DefaultConfigurationConstants.OPEN_ORDERS_SCHEDULER_PERIOD);
-			Long schedulerPeriod = Long.valueOf(schedulerPeriodStr);
-			this.attendOpenOrdersExecutor.scheduleAtFixedRate(new TimerTask() {
-				@Override
-				public void run() {
-					try {
-						attendOpenOrders();
-					} catch (Throwable e) {
-						LOGGER.error("Error while checking and submitting open orders", e);
-					}
-				}
-			}, 0, schedulerPeriod);
-		}
+	public ManagerController(Properties properties, InstanceProvider localInstanceProvider,
+			InstanceProvider remoteInstanceProvider) {
+		this.localMemberId = properties.getProperty(ConfigurationConstants.XMPP_ID_KEY);
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+
+		OpenProcessor openProcessor = new OpenProcessor(this.localInstanceProvider,
+				this.remoteInstanceProvider, this.localMemberId, properties);
+		this.openProcessorThread = new Thread(openProcessor);
+		
+		this.startManagerThreads();
 	}
 
 	/**
-	 * Method that try to get an Instance for an Open Order.
+	 * This method starts all manager threads, if you defined a new manager
+	 * operation and this operation require a new thread to run, you should
+	 * start this thread at this method.
 	 */
-	private void attendOpenOrders() {
-		LOGGER.info("Trying to get Instances for Open Orders");
-		List<Order> openOrders = this.managerDatastore.getOrderByState(OrderState.OPEN);
-
-		/**
-		 * TODO: this method can generate a race condition. For example:
-		 * a user can delete a Open Order while this method is trying to get an
-		 * Instance for this Order.
-		 */
-		for (Order order : openOrders) {
-			try {
-				order.handleOpenOrder();
-				OrderInstance orderInstance = null;
-				if (order.isLocal()) {
-					orderInstance = this.localInstanceProvider.requestInstance(order);
-				} else if (order.isRemote()) {
-					orderInstance = this.remoteInstanceProvider.requestInstance(order);
-				}
-				if (!orderInstance.getId().trim().isEmpty()) {
-					order.setOrderState(OrderState.SPAWNING);
-				} else {
-					throw new RuntimeException("OrderInstance Id not generated");
-				}
-			} catch (Exception e) {
-				LOGGER.error("Error while trying to get an Instance for Order: " + System.lineSeparator() + order, e);
-				order.setOrderState(OrderState.FAILED);
-			}
-			this.managerDatastore.updateOrder(order);
-		}
+	private void startManagerThreads() {
+		LOGGER.info("Starting Manager Threads...");
```

Comment:
I think it is better to have a log line to each thread started instead of this generic line

Reply:
Agreed.

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/ef9302e5cd2c4af39ab626841ed293e681623723).